src/bls/bls_signer.cpp

```diff
@@ -0,0 +1,109 @@
+#include "bls_signer.h"
+#include "bls_utils.h"
+#include "logging/oxen_logger.h"
+#include "crypto/keccak.h"
+#include <oxenc/hex.h>
+#include "ethyl/utils.hpp"
+#include <fstream>
+#include <fmt/core.h>
+
+static auto logcat = oxen::log::Cat("bls_signer");
+
+BLSSigner::BLSSigner(const cryptonote::network_type nettype, fs::path key_filepath) {
+    initCurve();
+    const auto config = get_config(nettype);
+    chainID = config.ETHEREUM_CHAIN_ID;
+    contractAddress = config.ETHEREUM_REWARDS_CONTRACT;
+    // This init function generates a secret key calling blsSecretKeySetByCSPRNG
+    secretKey.init();
+    if (fs::exists(key_filepath)) {
```

Comment:
There are some helper functions in the codebase for loading in keys which I've updated this code to use which has a more paranoid/rigorous process for dealing with secrets.

Updated at https://github.com/oxen-io/oxen-core/commit/82d585b194b4c026d4cbf2d6f6eec62dc6a07240